### DISCUSSION TITLE
pygtk install 404

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#2288] [Everyday usage]
When I try: brew install pygtk

I get the follow output:

==> Downloading https://ghcr.io/v2/homebrew/core/pygobject/manifests/2.28.7_2
#=#=#                                                                         
curl: (22) The requested URL returned error: 404 
Error: pygtk: Failed to download resource "pygobject_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/pygobject/manifests/2.28.7_2

Any help is appreciated!

**Comment by carlocab** [comment]
Version `2.28.7_2` is no longer in homebrew/core. What does `brew config` and `brew doctor` say?

**Reply by ghost** [reply]
**brew config:**
HOMEBREW_VERSION: 3.2.16-31-g976f9da
ORIGIN: https://github.com/Homebrew/brew
HEAD: 976f9daa12e86c9e1ff5ea88e95f3988a9855679
Last commit: 8 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f11a58264787268288b90f7e93ceb470b95e3819
Core tap last commit: 58 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: octa-core 64-bit ivybridge
Clang: 12.0.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4

**brew doctor**
Your system is ready to brew.

**Reply by carlocab** [reply]
Nothing out of the ordinary there. What about `brew info pygtk` and `brew info pygobject`?

**Reply by ghost** [reply]
Thanks for the help!

**brew info pygtk**

ryan-robeson/gimp/pygtk: stable 2.24.0 (bottled)
GTK+ bindings for Python
http://www.pygtk.org/
Not installed
From: https://github.com/ryan-robeson/homebrew-gimp/blob/HEAD/Formula/pygtk.rb
==> Dependencies
Build: pkg-config ✔
Required: atk ✔, glib ✔, gtk+ ✔, libglade ✔, py2cairo ✘, ryan-robeson/gimp/pygobject ✘

**brew info pygobject**

ryan-robeson/gimp/pygobject: stable 2.28.7 (bottled)
GLib/GObject/GIO Python bindings for Python 2
https://wiki.gnome.org/Projects/PyGObject
Not installed
From: https://github.com/ryan-robeson/homebrew-gimp/blob/HEAD/Formula/pygobject.rb
==> Dependencies
Build: pkg-config ✔
Required: glib ✔, ryan-robeson/gimp/python@2 ✔

**Reply by cho-m** [reply]
The tap you are using (https://github.com/ryan-robeson/homebrew-gimp) has copied over old formula's bottle section, which points to the no-longer-available default `homebrew-core` packages URL. These were probably removed back when formula was deleted in https://github.com/Homebrew/homebrew-core/commit/fcfa254fcf5152d85c1a3a058a8b41808c9a67e6

You may need to build this from source via `brew install -s <formula>`.

Or, request that external tap (or find another tap, or set up your own) to provide its own bottles using `root_url` (https://docs.brew.sh/Bottles#root-url-root_url). It looks like that tap has done this for some of their other formulae.
